### DISCUSSION TITLE
[Error] Regex could not be compiled

### DISCUSSION THREAD

**Discussion by chook100** [DSJAS/DSJAS#60] [Installation help] [answered]
Hello. I am trying to install DSJAS for the first time and keep getting this error in Apache error.log: 
`[Fri Dec 09 18:51:03.573705 2022] [core:alert] [pid 15284:tid 1856] [client ::1:50281] C:/xampp/htdocs/admin/.htaccess: Regex could not be compiled`

I have installed  XAMPP then I downloaded 0.1.2 Beta of DSJAS, unzipped it, and copied all 10 files straight into C:/xampp/htdocs (making sure to delete all previous files in this directory). I then started both Apache and MySQL and then went to "localhost" on Chrome. Upon going to local host, I was met with a 500 Internal Server Error. The URL was "http://localhost/admin/install/install.php" and the footer was Apache/2.4.54 (Win64) OpenSSL/1.1.1p PHP/8.1.12 Server at localhost Port 80.

Is this something I am overlooking?

**Comment by chook100** [comment, marked as answer]
Edit: I have solved the problem.
In this case, the regular expression is "\installTheme.php$", which is trying to match a file named installTheme.php at the end of the line. However, the backslash before installTheme.php is being treated as an escape character, so the regular expression is actually looking for a file named installTheme.php. I fixed this error by removing the backslash before installTheme.php, so that the regular expression becomes "installTheme.php$". This will match a file named installTheme.php at the end of the line, as intended.

For anyone having this issue, just replace all the code in C:/xampp/htdocs/admin/.htaccess to
```
# Allow larger uploads when installing themes
<Files ~ "installTheme.php$">
php_value upload_max_filesize 50M
php_value post_max_size 42M
</Files>

# Allow larger uploads when installing modules
<Files ~ "installTheme.php$">
php_value upload_max_filesize 50M
php_value post_max_size 42M
</Files>
```

**Reply by Muyu111** [reply]
Hi there, 
I have no experience with coding but followed your advice and it worked! Thank you so much for your guidance!

**Comment by TheUnsocialEngineer** [comment]
I've had this issue before and still have it now yet cant remember the fix I used however I do not have the aforementioned  "\installTheme.php$" line in my htaccess file and cant seem to figure out the issue I've attached my htaccess below if anyone know how to fix that would be very much appreciated .
[hta.txt](https://github.com/DSJAS/DSJAS/files/10326909/hta.txt)

**Reply by ejv2** [reply]
The issue was present in the admin .htaccess file. I have just pushed a quick patch to master, but if you want to fix it yourself, the file you want is in the "admin" subdirectory.

**Comment by ejv2** [comment]
Fixed as of commit 2058619.
